cardano-cli/src/Cardano/CLI/Shelley/Run/Address.hs

```diff
@@ -181,3 +182,21 @@ readAddressVerificationKeyFile (VerificationKeyFile vkfile) =
 runAddressBuildMultiSig :: ExceptT ShelleyAddressCmdError IO ()
 runAddressBuildMultiSig =
     liftIO $ putStrLn ("runAddressBuildMultiSig: TODO")
+
+--
+-- Script addresses
+--
+
+runAddressBuildScript
+  :: ScriptFile
+  -> NetworkId
+  -> Maybe OutputFile
+  -> ExceptT ShelleyAddressCmdError IO ()
+runAddressBuildScript (ScriptFile fp) nId mOutFp = do
+  script <- firstExceptT ShelleyAddressCmdReadFileError
+              . newExceptT $ readFileTextEnvelope AsScript fp
+  let payCred = PaymentCredentialByScript $ scriptHash script
+      scriptAddr = serialiseAddress $ makeShelleyAddress nId payCred NoStakeAddress
+  case mOutFp of
+    Just (OutputFile oFp) -> liftIO $ Text.writeFile oFp scriptAddr
```

Comment:
@duncan Are we purposely not writing the `Address Shelley` in the `TextEnvelope` format?

Reply:
Yes. Addresses have their own standard presentation format.